robottelo/config/__init__.py

```diff
@@ -69,4 +104,73 @@ def setting_is_set(option):
     opt_inst = getattr(settings, option, None)
     if opt_inst is None:
         raise ValueError(f'Setting {option} did not resolve in settings')
-    return opt_inst.validate() == [] or isinstance(opt_inst, DynaBox)
+    if hasattr(opt_inst, 'validate'):
+        return opt_inst.validate()
+    return isinstance(opt_inst, DynaBox)
+
+
+def configure_nailgun():
+    """Configure NailGun's entity classes.
+
+    Do the following:
+
+    * Set ``entity_mixins.CREATE_MISSING`` to ``True``. This causes method
+        ``EntityCreateMixin.create_raw`` to generate values for empty and
+        required fields.
+    * Set ``nailgun.entity_mixins.DEFAULT_SERVER_CONFIG`` to whatever is
+        returned by :meth:`robottelo.helpers.get_nailgun_config`. See
+        ``robottelo.entity_mixins.Entity`` for more information on the effects
+        of this.
+    * Set a default value for ``nailgun.entities.GPGKey.content``.
+    """
+    from nailgun import entities
+    from nailgun import entity_mixins
+    from nailgun.config import ServerConfig
+
+    entity_mixins.CREATE_MISSING = True
+    entity_mixins.DEFAULT_SERVER_CONFIG = ServerConfig(get_url(), get_credentials(), verify=False)
+    gpgkey_init = entities.GPGKey.__init__
+
+    def patched_gpgkey_init(server_config=None, **kwargs):
+        """Set a default value on the ``content`` field."""
+        gpgkey_init(settings, server_config, **kwargs)
```

Comment:
This is raising an exception:

```
tests/foreman/api/test_multiple_paths.py:232: in test_positive_get_status_code
    entity = entity_cls(id=entity_cls().create_json()['id'])
robottelo/config/__init__.py:124: in patched_gpgkey_init
    gpgkey_init(settings, server_config, **kwargs)
E   TypeError: patched_gpgkey_init() takes from 0 to 1 positional arguments but 2 were given
```

Reply:
Note this exception was seen on commit c552449, which is a bit out of date.

https://github.com/SatelliteQE/robottelo/commit/c55244941625cd8e0dcb80af8566c8a74a43b18b

Reply:
Still seeing this in https://github.com/SatelliteQE/robottelo/commit/02605813e3f2e97598c2a667b256229cc804df2a